### ISSUE TITLE
Scribus's status on github

### ISSUE THREAD

**Issue by OceanWolf** [scribusproject/scribus#22]
Hi people, my first time over here on the ``scribusproject`` account so a bit confused as to the goings on. and really that marks the centre of this issue.  Making sure other people know what goes on.

I have read http://scribus.io/ where it suggests I drop a message here.  I also see on that page it says:
> Git has also seemed to surpass Subversion

Does that mean that Github has become the main-home for Scribus now, but on the about of this repository, it describes this as
> Mirror of official Scribus SVN (http://bugs.scribus.net ← Submit PRs & Bugs) http://scribus.io

combine this with the lack of mention on the main scribus website, and I feel a bit confused... I have the sense that 

Not really a C coder myself, but I do collaborate on quite a large python project on Github, matplotlib, with 30 core developers including now myself and almost 400 contributors, so if I can help a bit let me know.  I love scribus and I finally start to see it coming of age, finding I can use it more and more.

**Comment by LiamHD** [comment]
+1 from my side. I also find it quite confusion.

**Comment by aoloe** [comment]
hey @OceanWolf the http://scribus.io site is not much more than a draft for now. well, a placeholder!

i have created a pad, where we can produce a better content for it:

https://graphicslab.titanpad.com/65

feel free to discuss its content in here and/or edit the pad.

i think that the main points on the page should be:

- scribus.io is a community project run by some of the most active contributors to scribus
- scribus.io is not an official scribus page and is not run nor blessed by the scribus team (but they know about its existence and have not said to stop)
- explain what we provide on github
  - a repository to be cloned
  - a repository to be forked
  - a way to make pull requests that can be discussed before being sent over as a patch on the bug tracker
  - a bug tracker for people that are too lazy and don't want to create an account on scribus' mantis bug tracker
  - a short introduction on how to work with the scribus code on github

do you (plural!) see any point that should be mentioned there?
something that does not belong there?

@kunda (the author of the current page): any thoughts?

**Comment by OceanWolf** [comment]
Okay, so basically this acts as a fork of the main scribus repo, with interesting stuff getting pushed back to the main scribus repo.  Otherwise we might add stuff that then doesn't get accepted by the scribus team.  It feels a bit disjointed otherwise trying to have two separate versions...

Right now I see we need two things:
1. A ``Readme.rst`` file with the information pertaining to this github fork.  People will most likely see that first before clicking through to the page.
2. HTML generated documentation of the code, over on matplotlib, we the entire main website gets generated from the contents of the git repo.

I have just forked scribus and will begin working on 1.  Not sure about how to build scribus or the doc, but I will leave that as a gap in the PR and can work on details like that there.

**Comment by aoloe** [comment]
eh eh... no, it's not a fork, it's a mirror.
if one day interesting pull requests don't make it to the trunk, we will have to think about making part of this some sort of fork. but for now we don't have such patches so...

for the two other things:

1. please call it README_github.md ... if possible. I will create the file later this evening and check if it survives our svn to github update script! thanks for working on it!
2. i would say that relevant parts of the the scribus code are not well enough structured and documented to make a doxygen documentation urgent... but it could inspire people to make patches that improve it, once we have it :-)
  i  have added a ticket to the website repository (https://github.com/scribusproject/scribusproject.github.io/issues/3) and i will possibly make a try, if nobody else does it (i wanted to produce the documentation for the scribus plugin API i have created... so i might reuse my in-progress skills)

**Comment by OceanWolf** [comment]
1. Almost finished typing it up, not much difference between rst and md, so all good, any reason why you prefer md to rst?
2. Yes, I find one of the best ways to learn about code comes from writing documentation, and you leave behind something for others, making it easier for them to learn.

**Comment by luzpaz** [comment]
I started writing this last night but didn't get a moment till now to send. 
I think @aoloe did a good job summarizing the current situation of Scribus on github. 
I will add some of what I wanted to write yesterday.

-------

Hey @OceanWolf!  
Welcome to ```scribusproject```. Definitely appreciate your feedback and questions. I don't represent Scribus and won't speak on behalf of it. I'm simply a volunteer 

>>    Git has also seemed to surpass Subversion  

> Does that mean that Github has become the main-home for Scribus now, but on the about of this repository?

No, the core devs are only using the SVN. When any PRs are submitted to the Scribus Github mirror one of the volunteers will make a submission to Scribus bugtracker. It's not efficient or ideal but it's a way to have a presence on Github. Specifically for interactions like this one. Matplotlib is a fascinating project that I have yet to explore. As you mentioned is has a battalion of developers and contributors. That is an exciting to see, a thriving open source project.

**Comment by luzpaz** [comment]
I'm pretty excited about this thread. I'm glad for any feedback and revisions about scribus.io. It needs a lot of love. I was using a jekyll template for scribus.io that github provides but I don't have much love for it. It would be nice to redo it with the option of adding a blog and other cool functionality that could help others orient around Scribus development. 

Lets talk some more about:
```2. HTML generated documentation of the code, over on matplotlib, we the entire main website gets generated from the contents of the git repo.```

**Comment by OceanWolf** [comment]
The main non-md code I wrote were
```
.. NOTE::
    This message appears in a box called Note
```

I haven't explored markdown much, it just seems like a more basic version of reStructuredText, and yes the entire site gets generated from restructured text files, also because the docstrings get written in reStructuredText, as it supports table of contents, and allows us to split documentation up amongst many files, and reST compiles it all together.  I have only joined the matplotlib "core-developer" team recently though, earlier this year, I put that in brackets, as git differentiates between "core-developers" and "owners", where as above @aoloe referred to the owners here as the "core-developers", so a bit of a lingo mish-mash.

**Comment by aoloe** [comment]
@OceanWolf personally i prefer markdown because it has a simpler syntax, with very little overhead over pure text.
and forces the writer to avoid complex formatting :-)
(well, i have to admit that i'm using a few extensions to markdown in my files...)

**Comment by aoloe** [comment]
concerning the different "types" of developpers:

- there is a Scribus team and it's a well defined group of people. that's the official Scribus.
- there is a Scribus community which is a more variable group of people. some of the people in the community have been committing lot of time to scribus over many years. others are (or have been) giving lot of contributions for a limited time with much dedication and are not in them. some have been much involved in scribus for many years and are not in the team.

currently, three members of the team can also be defined as "the core developers", regularly squashing bugs and managing the patches that are proposed. i would avoid that term on the scribus.io site.

i hope that i'm not making things ever more complicated :-)

**Comment by OceanWolf** [comment]
Lol, too complicated :wink: the Scribus team should just migrate over to github, much easier, maybe if we get enough contributors they will :innocent:.  (Or gitlab, gitlab works too ;)).

**Comment by LiamHD** [comment]
Do you now that github repos can be cloned as svn repos? See https://github.com/blog/966-improved-subversion-client-support
I don't know how good this support is but maybe this is a way for the core devs to still use svn.

**Comment by aoloe** [comment]
all this has been discussed lot of times.
if the team wants to switch to git or github they will make a sign, don't worry.
in my eyes, the only thing that can help, is having an healthy crew of contributors here on github so that they get a figure of what they are missing.
at this point, words won't be of any help... they are comfortable with their svn workflow and don't have any need  to switch.
you see a reason to switch, but are not (yet) contributing any code...
i see a reason to switch, but they know that i can keep on working with svn as i've done in the past :-) (and i don't contribute much code either)
if one day we will have a few regular contributors that are on github and are pissed off by the over complicated workflow, we restart the discussion!
for now, i'd prefer to mention this idiosyncrasy but at the same time we should  try not to give too much weight to it.

**Comment by OceanWolf** [comment]
hehe, exactly what I tried to say, albeit in a lighthearted tongue-in-cheek way.

**Comment by LiamHD** [comment]
@aoloe D'accord.
I just wanted to provide all the possible options.

**Comment by luzpaz** [comment]
@OceanWolf @LiamHD just want to say thank you so far for the feedback and contributions. @aoloe and myself spend a lot of time spreading the word about Scribus. We have a sense that github has immense possibilities and like I said before threads like this one are exciting because it shows people are interested and want to contribute. 

We really want to see Scribus thrive more. Believe me, it's been gaining momentum. As this is occurring we're working with the pace of that. So thank you for your understanding and patience.

**Comment by OceanWolf** [comment]
Definitely, I remember a few years ago when I first tried Scribus, hearing it as the opensource DTP package of choice, and got turned heavily turned off by its lack of stability, it just kept on wanting to crash every few seconds.

Last year I took another look getting back into DTP, and found the stability issues fixed and used it for a newsletter, a flyer, and couple of posters :smile:.  I then felt for the first time that Scribus had almost come of age, a year or two off (touch wood) from becoming on par with the big name applications like Inkscape.

Right now, I really look forward to giving 1.5 a whirl and getting the documentation site up and running, examples, and some unit testing for travis to sink its teeth into, oh and learning how to use python with scribus... I don't seem to see any python.  With the talk regarding git-svn problems though examples might become a bit difficult...

Basically I envisage migrating a lot of the "tutorials" in the wiki into the trunk... nothing I find more discouraging than trying to learn a programme and finding a lot of out of date tutorials, followed second by an inconsistent tutorial series.  Because of this I still feel quite noobish and scared with Scribus.

Anyone one step at a time... lets get the basics set up to allow people to use and contribute to Scribus.

My current question, how do I run the python?
```python
from scribus import *
```
I can't the module anywhere in this repository...

**Comment by aoloe** [comment]
you can only run python scribus inside of scribus... does this answer your question? :-)

**Comment by OceanWolf** [comment]
Erm yes, still confused, but yes :).
